src/output/plugins/httpd/HttpdClient.cxx

```diff
@@ -41,6 +47,10 @@ HttpdClient::BeginResponse() noexcept
 {
 	assert(state != State::RESPONSE);
 
+	if (!head_method) {
+	       should_reject_unauthorized = (strlen(httpd.password) > 0 && (!provided_password || (strcmp(provided_password->c_str(), httpd.password) != 0)));
```

Comment:
The `strlen()` call is clumsy and adds overhead, because it needs to walk the whole string until the end for no reason. All you want to know is whether the string is empty, and `*http.password != 0` tells you that without walking the string. MPD has a convenience function for that to make it easier to understand: `!StringIsEmpty(http.password)` (does the same, but gives it a name which declares your intention).
But if you change it according to my other hints, you can easily write instead:
```
should_reject_unauthorized = http.password != nullptr && provided_password != http.password;
```
Simple, eh?

Reply:
Yeah, nice. Done.